pyanaconda/payload/dnfpayload.py

```diff
@@ -912,6 +912,10 @@ def environmentDescription(self, environmentid):
 
     def environmentId(self, environment):
         """Return environment id for the environment specified by id or name."""
+        # the enviroment must be string or else DNF throws an assert error
+        if not isinstance(environment, str):
+            log.warning("environmentId() called with non-string argument: %s", environment)
+            raise TypeError
```

Comment:
If the environment is not `None` or a string, then it is a bug in Anaconda. There is no reason to raise an exception, that will be later ignored. It is ok, that DNF fails in this case and it will cause Anaconda to crash.

Reply:
I'll remove the TypeError then will leave the warning & comment so that any issues can be more easily debugged (the assert error looks pretty confusing at first glance).